config/initializers/content_security_policy.rb

```diff
@@ -1,26 +1,91 @@
-    policy.style_src   :self, :https
-    # Specify URI for violation reports
-    # policy.report_uri "/csp-violation-report-endpoint"
-  end
+GOVUK_DOMAINS = [
+  '*.education.gov.uk',
+  '*.s3.eu-west-2.amazonaws.com',
+  'eyfs-cms-preprod.london.cloudapps.digital',
+].uniq.freeze
+
+GOOGLE_ANALYTICS_DOMAINS = %w[www.google-analytics.com
+                              ssl.google-analytics.com
+                              stats.g.doubleclick.net
+                              www.googletagmanager.com
+                              *.ytimg.com
+                              www.youtube.com
+                              www.youtube-nocookie.com
+                              *.hotjar.com].freeze
+
+OPTIMIZE_DOMAINS = %w[www.googleoptimize.com
+                      optimize.google.com
+                      fonts.googleapis.com].freeze
 
-  # Generate session nonces for permitted importmap and inline scripts
-  config.content_security_policy_nonce_generator = ->(request) { request.session.id.to_s }
-  # config.content_security_policy_nonce_directives = %w(script-src)
+GOOGLE_STATIC_DOMAINS = %w[fonts.gstatic.com www.gstatic.com *.hotjar.com].freeze
 
-  # Report CSP violations to a specified URI. See:
-  # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
-  # config.content_security_policy_report_only = true
+Rails.application.config.content_security_policy do |policy|
+  policy.default_src :self, :https, *GOVUK_DOMAINS
+  policy.font_src    :self, :https, *GOVUK_DOMAINS, *GOOGLE_STATIC_DOMAINS, :data
+  policy.frame_src   :self, *GOOGLE_ANALYTICS_DOMAINS, *OPTIMIZE_DOMAINS
+  policy.img_src     :self,
+                     *GOVUK_DOMAINS,
+                     *GOOGLE_ANALYTICS_DOMAINS, # Tracking pixels
+                     *OPTIMIZE_DOMAINS,
+                     :data # Base64 encoded images
+  policy.object_src  :none
+  policy.script_src  :self,
+                     *GOVUK_DOMAINS,
+                     *GOOGLE_ANALYTICS_DOMAINS,
+                     *GOOGLE_STATIC_DOMAINS,
+                     *OPTIMIZE_DOMAINS,
+                     # Allow YouTube Embeds (Govspeak turns YouTube links into embeds)
+                     '*.ytimg.com',
+                     'www.youtube.com',
+                     'www.youtube-nocookie.com',
+                     # Allow all inline scripts until we can conclusively
+                     # document all the inline scripts we use,
+                     # and there's a better way to filter out junk reports
+                     :unsafe_inline
+  policy.style_src   :self, *GOVUK_DOMAINS, *GOOGLE_STATIC_DOMAINS, *OPTIMIZE_DOMAINS, :unsafe_inline
+  if Rails.env.development?
+    # :nocov: by definition will not run in test
+    policy.connect_src :self,
+                       :https,
+                       :wss,
+                       *GOVUK_DOMAINS,
+                       *GOOGLE_ANALYTICS_DOMAINS,
+                       'http://localhost:3035',
```

Comment:
What is this port for?

Reply:
Not Sure it's a reform thing I think, that was there originally I can remove it if docker is not reliant on it.